src/objective-c/tests/UnitTests/APIv2Tests.m

```diff
@@ -770,4 +843,319 @@ - (void)testFlowControlReadNonBlockingFailure {
+  __weak XCTestExpectation *metadata1 =
+      [self expectationWithDescription:@"Received initial metadata for RPC1."];
+  __weak XCTestExpectation *metadata2 =
+      [self expectationWithDescription:@"Received initial metadata for RPC2."];
+  NSArray *initialMetadataDone = [NSArray arrayWithObjects:metadata1, metadata2, nil];
+
+  RMTStreamingOutputCallRequest *request = [RMTStreamingOutputCallRequest message];
+  RMTResponseParameters *parameters = [RMTResponseParameters message];
+  parameters.size = kSimpleDataLength;
+  [request.responseParametersArray addObject:parameters];
+  request.payload.body = [NSMutableData dataWithLength:kSimpleDataLength];
+
+  GRPCRequestOptions *requestOptions =
+      [[GRPCRequestOptions alloc] initWithHost:kHostAddress
+                                          path:kFullDuplexCallMethod.HTTPPath
+                                        safety:GRPCCallSafetyDefault];
+  GRPCMutableCallOptions *callOptions = [[GRPCMutableCallOptions alloc] init];
+  callOptions.transportType = GRPCTransportTypeInsecure;
+  GRPCCall2 *call1 = [[GRPCCall2 alloc]
+      initWithRequestOptions:requestOptions
+             responseHandler:[[ClientTestsBlockCallbacks alloc]
+                                 initWithInitialMetadataCallback:^(NSDictionary *initialMetadata) {
+                                   [metadata1 fulfill];
+                                 }
+                                 messageCallback:nil
+                                 closeCallback:^(NSDictionary *trailingMetadata, NSError *error) {
+                                   [completion1 fulfill];
+                                 }]
+                 callOptions:callOptions];
+  GRPCCall2 *call2 = [[GRPCCall2 alloc]
+      initWithRequestOptions:requestOptions
+             responseHandler:[[ClientTestsBlockCallbacks alloc]
+                                 initWithInitialMetadataCallback:^(NSDictionary *initialMetadata) {
+                                   [metadata2 fulfill];
+                                 }
+                                 messageCallback:nil
+                                 closeCallback:^(NSDictionary *trailingMetadata, NSError *error) {
+                                   [completion2 fulfill];
+                                 }]
+                 callOptions:callOptions];
+  [call1 start];
+  [call2 start];
+  [call1 writeData:[request data]];
+  [call2 writeData:[request data]];
+  [self waitForExpectations:initialMetadataDone timeout:kTestTimeout];
+  GRPCCall2Internal *internalCall1 = call1->_firstInterceptor;
+  GRPCCall2Internal *internalCall2 = call2->_firstInterceptor;
+  XCTAssertEqual(
+      internalCall1->_call->_wrappedCall->_pooledChannel->_wrappedChannel->_unmanagedChannel,
+      internalCall2->_call->_wrappedCall->_pooledChannel->_wrappedChannel->_unmanagedChannel);
+  [call1 finish];
+  [call2 finish];
+  [self waitForExpectations:rpcDone timeout:kTestTimeout];
+}
+
+- (void)testChannelReuseDifferentPath {
```

Comment:
I didn't quite understand what this test is trying to test. Could you clarify (maybe with some comment on the top of this test)?

Reply:
It looks like you are trying to test calls with different paths use the same channel, correct? But the paths are currently the same.

Reply:
Oh sorry, the path's were the same so this test was useless. I've removed this testcase.